### PR TITLE
No Use* from stages

### PR DESCRIPTION
**Pull request by tokatoka**

### CODE REVIEW

**Comment by tokatoka** [comment]
It was impossible to remove some of UsesState because trait such as ExecutionProcess or EventFirer really needs it.

**Comment by tokatoka** [comment]
but events is heavily dependent on fuzzer and fuzzer is dependent on stage
so i could only start from here.

only after this can i touch other modules

**Comment by domenukk** [comment]
Also, it's a bit sad we get _more_ generics this way? Wasn't it the point of Uses* to get rid of some? I guess I asked this same question for every PR :D

**Comment by tokatoka** [comment]
>  it's a bit sad we get more generics this way?

it could be so because trait `State` has been changed with smaller sets of contraints,
and it could look so because for now i cannot completely remove UsesState while EventManager, Executor ... still use them

**Comment by tokatoka** [comment]
everytime i make pr i have more deleting so it's fine

File: fuzzers/forkserver/libafl-fuzz/src/stages/mutational_stage.rs [hunk 1]
```diff
@@ -1,36 +1,23 @@
...
-    E: UsesState<State = S>,
-    EM: UsesState<State = S>,
-    M: Mutator<I, S>,
-    Z: Evaluator<E, EM, State = S>,
-    I: MutatedTransform<S::Input, S> + Clone + Input,
-    SM: MutationalStage<E, EM, I, M, Z, State = S>,
-    P: MutationalStage<E, EM, I, M, Z, State = S>,
-    S: State<Input = I> + HasRand + HasCorpus + HasNamedMetadata,
-    <<Self as UsesState>::State as HasCorpus>::Corpus: Corpus<Input = Self::Input>, //delete me
+    SM: MutationalStage<M, S>,
```
Comment (domenukk) [inline fuzzers/forkserver/libafl-fuzz/src/stages/mutational_stage.rs:18]: Nice! <3

File: fuzzers/forkserver/libafl-fuzz/src/stages/mutational_stage.rs [hunk 2]
```diff
@@ -1,36 +1,23 @@
...
+    P: MutationalStage<M, S>,
 {
-    /// The mutator, added to this stage
```
Comment (domenukk) [inline fuzzers/forkserver/libafl-fuzz/src/stages/mutational_stage.rs:32]: why remove the comment

Comment (tokatoka) [reply fuzzers/forkserver/libafl-fuzz/src/stages/mutational_stage.rs:32]: mistake

**Comment by tokatoka** [comment]
> it's a bit sad we get more generics this way? 

regarding, this
we *will* have more generics but less traits bounds

**Pull request merged**